hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -91,16 +90,8 @@ boolean contains(String name) {
      * @param result The ProxyInfo of all proxies in this registry.
      */
     public void getProxyInfos(Collection<ProxyInfo> result) {
-        for (Map.Entry<String, DistributedObjectFuture> entry : proxies.entrySet()) {
-            final DistributedObjectFuture future = entry.getValue();
-            if (!future.isSet()) {
```

Comment:
Right it will return `false`, only wanted to sketch an example scenario but in any case, with removal of `future.isSet()`, post-join operation may wait unneededly longer on joining node and timed out post-join operations can be seen more often (light weight post joins preferred generally)

To fix this specific issue, isn't it sufficient to remove `instance of InitializingObject` check? Do we need to remove`future.isSet()` also?